packages/realm/src/app-services/App.ts

```diff
@@ -65,12 +70,13 @@ export class App {
 
   public userAgent = `RealmJS/${App.SDK_VERSION} (${App.PLATFORM_CONTEXT}, ${App.PLATFORM_OS}, v${App.PLATFORM_VERSION})`;
 
-  private listeners = new Listeners<AppChangeCallback, unknown>({
-    register() {
-      throw new Error("Not yet implemented");
+  private listeners = new Listeners<AppChangeCallback, AppListenerToken, [binding.App]>({
+    register(callback: () => void, internal: binding.App): AppListenerToken {
+      const token = internal.subscribe(callback);
+      return { internal, token };
```

Comment:
Sorry for the confusion on this when we went through it on the call. It shouldn't be needed to include the App internal into the token, since you can simple read it off the SDK `App` instance using `this.internal`.

```suggestion
  private listeners = new Listeners<AppChangeCallback, binding.AppSubscriptionToken>({
    register(callback: () => void): binding.AppSubscriptionToken {
      return this.internal.subscribe(callback);
```

Reply:
This is fixed now.  Needed to convert this into an arrow function so that the `App` context was binded to `this`.